src/CartsList.tsx

```diff
@@ -0,0 +1,164 @@
+    } else {
+      setSelectedCarts([])
+    }
+  };
+
+  const modalRef = useOnclickOutside(() => {
+    setIsShowModal(false);
+  });
+
+  const onDeleteCart = () => {
+    const promises = selectedCarts.map(el => removeCartItems(el));
+    Promise.all(promises)
+      .then(() => {
+        updateCartData();
+        setIsShowModal(false);
+        setSelectedCarts([]);
+      })
+      .catch(error => {
+        console.error(error);
+      });
+  };
+
+  const handleCart = (cart:any) => {
+    updateSelectedCartName(cart.name);
+    onSelectCart(cart);
+    onHandlePage('itemList');
+    localStorage.setItem('mcart', cart.id);
+    setIsCartSelected(true)
+  };
+
+  return (
+    <div className="cartslist">
+      {multiCartData.length && (
+        <div role="presentation" className="cartslist__editbutton" onClick={() => setIsEdit(!isEdit)}>
+          {t(isEdit ? 'done' : 'edit')}
+        </div>
+      )}
+      <div>
+        <h2 className="cartslist__title">
+          {t('my-carts')}
+        </h2>
+        {multiCartData && multiCartData.length ? (
+          <div>
+            <div className="cartslist__selectedtitle">
+              <div className={`${isEdit && "isshow"}`}>
+                <span>
+                  <input type="checkbox" name="cartCheck" id="select-all" className="cartslist__checkall epcheckbox" onChange={() => {handleSelectAll()}} />
+                  <label htmlFor="select-all" className="">
+                    {selectedCarts.length}
+                    &nbsp;
+                    {t('selected')}
+                  </label>
+                </span>
+                <button className="cartslist__deletebutton" disabled={selectedCarts.length === 0} onClick={() => setIsShowModal(true)}>
+                  <DeleteIcon />
+                </button>
+              </div>
+            </div>
+            <div className="cartslist__cartlist">
+              {multiCartData.map((cart: any) => (
+                <div className={`cartslist__cartelement ${cart.id.includes(selectedCartData.id) && "--selected"}`} key={cart.id}>
+                  {isEdit && (
+                    <input type="checkbox" name="cartCheck" id={`cart_${cart.id}`} className="cartslist__check epcheckbox" checked={selectedCarts.includes(cart.id)} onChange={() => {handleSelectCart(cart.id)}} />
+                  )}
+                  <label htmlFor={`cart_${cart.id}`} className="cartslist__description">
+                    <div className="cartslist__cartname">
+                      <strong className="cartslist__name">
+                        {cart.name}
+                      </strong>
+                      <span className="cartslist__select">
+                        <span className="cartslist --date">
+                          {t('created')} - {cart.meta.timestamps.created_at}
+                        </span>
+                        <button className="cartslist__selectcart" onClick={() => handleCart(cart)}>
```

Comment:
We'll need to confirm with UX if the button on the arrow is sufficient for a click action. It's not as intuitive to click the arrow vs the entire cart list item
FYI @BonnieEP